contracts/btc-finality/src/queries.rs

```diff
@@ -1,22 +1,25 @@
 use crate::error::ContractError;
 use crate::msg::{
     ActiveFinalityProvidersResponse, BlocksResponse, EvidenceResponse,
-    FinalityProviderPowerResponse, FinalitySignatureResponse, JailedFinalityProvider,
-    JailedFinalityProvidersResponse, SigningInfoResponse, VotesResponse,
+    FinalityProviderPowerBatchResponse, FinalityProviderPowerResponse, FinalitySignatureResponse,
+    JailedFinalityProvider, JailedFinalityProvidersResponse, SigningInfoResponse, VotesResponse,
 };
 use crate::state::finality::{
     get_last_signed_height, get_power_table_at_height, BLOCKS, EVIDENCES, FP_START_HEIGHT, JAIL,
     NEXT_HEIGHT, SIGNATURES,
 };
 use babylon_apis::finality_api::IndexedBlock;
 use cosmwasm_std::Order::{Ascending, Descending};
-use cosmwasm_std::{Deps, StdResult};
+use cosmwasm_std::{Deps, StdError, StdResult};
 use cw_storage_plus::Bound;
 
 // Settings for pagination
 const MAX_LIMIT: u32 = 30;
 const DEFAULT_LIMIT: u32 = 10;
 
+// Maximum number of heights allowed in batch queries
+const MAX_HEIGHTS_LIMIT: usize = 100;
```

Comment:
```suggestion
// Maximum number of heights allowed in batch queries
const MAX_HEIGHTS_LIMIT: usize = 30;
```

to align with other limits. This is due to costs in CW operation

Reply:
Is 30 the upper limit? could we squeeze 50 😅

Reply:
we can try but not 100% sure

Reply:
<img width="1289" height="804" alt="image" src="https://github.com/user-attachments/assets/a3b38515-9f19-4100-876b-4c1816b070d7" />

Managed to return 100 from the contract. Is it okay to leave 100 in then